### ISSUE TITLE
Add structure to incoming support mail

### ISSUE THREAD

**Issue by garethrees** [mysociety/whatdotheyknow-theme#740]
An important task to tackle to reduce support burden is the reduction of inbound emails to the support inbox at source, and the pre-categorisation / allocation of emails as they come in.

A good place to add context is in the "My message is about" (i.e. the Subject) field.

![The-Culprit](https://user-images.githubusercontent.com/282788/101152238-ad54d680-361a-11eb-8524-501a0e5b1ae0.png)

* Rather than a free text Message title we should first assess the different categories of inbound message so that we have a common list of queries
* Instead of displaying the whole form we should look to build a step by step approach whereby you can select the problem type and where possible we actively direct people to specific help pages first of all, relevant to that issue
* Only once those options are exhausted will you get to a form where you can type in your message and submit
* These pre-categorised problems could then be automatically labelled in the inbox, and where relevant could be directed to specific staff or volunteers depending on the issue

- [ ] We should collect all places that someone can submit an inbound email to the support inbox
- [ ] We should figure out [common categories based on existing labels in the inbox](https://docs.google.com/spreadsheets/d/1niel3ZoBqAjaQpW3Ag3O3E2yVgWQ1YfrdZLXa6hIPzQ/edit?usp=drivesdk).

**Comment by HelenWDTK** [comment]
The contact form has now become the primary way that our users and public authorities get in contact with us. To improve our handling incoming reports, we should consider adding some more structure to these, that would allow us to automatically categorise them in our inbox, akin to how we manage reports made via the report links. This could be achieved by presenting users with a selection of topics that, when chosen, insert specific identifiers in the subject line indicating the nature of the issue. We could detect this text, and triage the report by auto labelling it.

The upcoming 'Report a Data Breach' form will significantly improve our handling of these, however, my ideal setup would involve being able to make a clear distinction between critical, time-sensitive, risky matters (those that would receive a !Risk label/GDPR requests), and routine inquiries such as site operation questions, basic FOI assistance, feature suggestions, and users seeking requests being submitted on their behalf.

Frequently, site operation inquiries and basic FOI questions can be addressed by directing users to a help page or ICO guidance. One approach could be to guide users through a series of questions about their query before presenting them with the contact form. This approach would be designed to resolve their query before they submit it, and could potentially reduce the volume of incoming inquiries.

In implementing these changes, it's important that we continue prioritise accessibility and overall user experience. We don't want the form to be challenging to navigate or annoying to use.

**Comment by mdeuk** [comment]
> Frequently, site operation inquiries and basic FOI questions can be addressed by directing users to a help page or ICO guidance. One approach could be to guide users through a series of questions about their query before presenting them with the contact form. This approach would be designed to resolve their query before they submit it, and could potentially reduce the volume of incoming inquiries.

That's the very one we discussed recently - I think a small proof of concept, using a tool such as [EnquiryWitch](https://github.com/geeksforsocialchange/enquirywitch), might be a good way forward.

**Comment by garethrees** [comment]
There are two steps here.

1. Implement a mechanism for auto-labelling queries
2. Implement a mechanism for solving queries before the form is submitted

## 1. Implement a mechanism for auto-labelling queries

Given we'd want the first of these anyway, we'll focus on that for now.

The first step we can take here is to add a field to the form that asks what the topic is:

![Screenshot 2023-06-07 at 15 58 45](https://github.com/mysociety/whatdotheyknow-theme/assets/282788/cc4d74c9-c1e8-4f19-862c-4b56f781e3ff)

@HelenWDTK We'll need a list of these topics in user-facing language, and what label they would map to.

> We could detect this text, and triage the report by auto labelling it

We've got a couple of options for this:

1. Use the subject line
2. Add to the body text
3. Use a [CC address like GitHub](https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#filtering-email-notifications) does.

What we pick here will probably depend on the list of topics.

## 2. Implement a mechanism for solving queries before the form is submitted

> Frequently, site operation inquiries and basic FOI questions can be addressed by directing users to a help page or ICO guidance.

Once we have a topic field in place, we can dynamically render a sidebar that provides the most relevant signposting for the query. This isn't as good as a Q&A system, but it's a more achievable first step.

![Screenshot 2023-06-07 at 16 10 57](https://github.com/mysociety/whatdotheyknow-theme/assets/282788/a25b75f6-3d5c-44b2-9c54-a380c3060877)

In terms of implementation, we could easily map the fields in the select box to a partial to render dynamically, e.g:

* `FOI Help` → `views/help/contact/_foi_help.html.erb`
* `Account Help` → `views/help/contact/_account_help.html.erb`
* etc…

> using a tool such as [EnquiryWitch](https://github.com/geeksforsocialchange/enquirywitch)

I think this is a bit far from where we are right now. This might be something to consider to integrate with https://github.com/mysociety/alaveteli/issues/7764.

**Comment by HelenWDTK** [comment]
> We'll need a list of these topics in user-facing language, and what label they would map to.

That would depend on how many options we'd want to usefully add/exactly how we'd phrase them eg topics vs questions.

So we could have:

"I'm having trouble using WhatDoTheyKnow" -> `Help/Basic site use`
"I've got a question about how WDTK works" -> `Site operation query`
"I've got a question about FOI" -> `Help/Basic FOI`
"I'd like someone to make a request for me" -> `seeking volunteer`

For risky things, we could have:
"I'd like to report a legal issue" -> `!Risk  `
"Someone is harassing me" ->` !Risk/Harrassment`
"Someone is pretending to be me" -> `Risk/Impersonation`
"You are publishing information that infringes my Copyright" -> `Risk/Copyright`
"You are publishing material that is defamatory" -> `Risk Defamation` etc

OR 

We could go more topic based eg have a heading "Legal issues" then bullets:
- Copyright -> `!Risk/Copyright`
- Defamation -> `!Risk/Defamation`
- Other -> `!Risk `
etc

**Comment by garethrees** [comment]
> My suggested approach would be to use a mail header…

Ah, I'd thought about a header but didn't think we could label based off that. Of course, the Monster is a clever monster! One advantage of the subject line approach is that it would be more useful for other Alavetelis, but I do like the cleanliness of a header not adding a load of "internal" stuff to the mail that ends up going back and forth with users.

I think let's start with the form location and topic (i'd call the more like `X-Alaveteli-Source` and `X-Alaveteli-Topic`, but details details) to get the framework of this in place, and then we can add the extra bits as incremental improvements once we've got the two pieces talking to one another.

**Comment by garethrees** [comment]
Next step is to answer the questions from [the above](https://github.com/mysociety/whatdotheyknow-theme/issues/740#issuecomment-1587505889).

**Comment by HelenWDTK** [comment]
WTT gives you options for queries that are best answered by some means other than contacting the team
<img width="744" alt="Screenshot 2023-06-16 at 11 53 30" src="https://github.com/mysociety/whatdotheyknow-theme/assets/120410992/543f61f7-0616-42d8-b2c1-de73095ce330">
FMS use radio buttons with limited options
<img width="652" alt="Screenshot 2023-06-16 at 11 53 51" src="https://github.com/mysociety/whatdotheyknow-theme/assets/120410992/cd8bb9e8-4b53-4d56-b7fd-618138b3a710">
TWFY doesn't have a form.

General mySociety points to each service, then has an option for selecting which one you're writing about:
<img width="834" alt="Screenshot 2023-06-16 at 11 58 17" src="https://github.com/mysociety/whatdotheyknow-theme/assets/120410992/1dd38325-5ba6-4c28-a23d-d4aa5c4e4a09">

**Comment by HelenWDTK** [comment]
> Does topic + question using a select with optgroups seem okay, given the above?

Yes, despite being out of sync with other services. The volume and diversity of correspondence that these receive will be much less than with WDTK

>  If so, what are the actual groups and questions we want to use?

Let's keep it simple to start with, and try something like.

Legal:
"Someone is harassing me" -> !Risk/Harrassment
"Someone is pretending to be me" -> Risk/Impersonation
"You are publishing information that infringes my Copyright" -> Risk/Copyright
"You are publishing material that is defamatory" -> Risk Defamation
"I have another legal issue" -> !Risk

FOI & EIR help
"I've got a question about FOI/EIR" -> Help/Basic FOI
"I'd like someone to make a request for me" -> admin action: volunteer makes request 

Technical
"I'm having trouble using WhatDoTheyKnow" -> help/Basic site use
"I've got a question about how WDTK works" -> help/site operation query
"I've got a great idea for a new feature" -> help/user-feedback 

Personal Information
"I'd like you to stop publishing my name" -> GDPR&DPA/Rt. to Object
"I'd like you to delete the information that you hold about me" -> GDPR&DPA/ Rt. to Erasure
"I have another query about your processing of my personal information" -> GDPR&DPA

Feel free to tweak the wording at will.
